behind-the-veil-siteroot/imports/ui/components/pages/review/Review.jsx

```diff
@@ -1,27 +1,358 @@
+        }
+    }
+
+
+    if (isLoading || serviceData == null) {
+        // is loader, display loader
+        return (
+            <WhiteBackground pageLayout={PageLayout.LARGE_CENTER}>
+                <Loader
+                    loadingText={"loading . . ."}
+                    isLoading={isLoading}
+                    size={100}
+                    speed={1.5}
+                />
+            </WhiteBackground>
+        );}
+    else {
+        return (
+            <WhiteBackground pageLayout={PageLayout.LARGE_CENTER}>
+                <PreviousButton/>
+                {/* main container for content */}
+                <div className="flex flex-col gap-4 xl:px-40">
+                    <div className="large-text">Leave a Review</div>
+                    {/* display review details as a card */}
+                    <ServiceDetailsHeader
+                    service={serviceData.serviceName}
+                    type={serviceData.serviceType}
+                    artist={artistData.profile.alias}
+                    price={serviceData.servicePrice}
+                    />
+                    <div>
+                        {/* if no review exists, show the leave review page. otherwise, show the review.*/}
+                        {leaveReview ? (
+                            <form onSubmit={handleSubmitReview} className="flex flex-col gap-6">
+                                <div className="flex flex-col md:flex-row gap-4 md:gap-10">
+                                    {/* review title input */}
+                                    <div>
+                                        <Input 
+                                            id={reviewTitleInputId} 
+                                            label={
+                                                <label
+                                                    htmlFor={reviewTitleInputId}
+                                                    className="main-text text-our-black"
+                                                >
+                                                    Review Title
+                                                </label>
+                                            }
+                                            type="text"
+                                            placeholder="Enter Review Title"
+                                            className={"w-64 sm:w-96 lg:w-64 xl:w-96"}
+                                            name="reviewTitle"
+                                            onChange={(e) => {
+                                                setReviewTitle(e.target.value)
+                                            }}
+                                        />
+                                        {!reviewTitle && errors.inputReviewTitle && <span className="text-cancelled-colour">{errors.inputReviewTitle}</span>}
+                                    </div>
+                                    {/* review rating input */}
```

Comment:
this doesn't actually update so it keeps throwing an error saying this review doesnt have a title